src/lib/eap_aka_sim/encode.c

```diff
@@ -97,18 +99,18 @@ static bool is_eap_aka_encodable(void *item, void *uctx)
 	+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
  @endverbatim
  */
-static ssize_t encode_iv(uint8_t *out, size_t outlen, void *encoder_ctx)
+static ssize_t encode_iv(fr_dbuff_t *dbuff, void *encoder_ctx)
 {
 	fr_aka_sim_encode_ctx_t	*packet_ctx = encoder_ctx;
-	uint8_t			*p = out;
 	uint32_t		iv[4];
+	fr_dbuff_t		work_dbuff = FR_DBUFF_NO_ADVANCE(dbuff);
 
 	/*
 	 *	One IV per packet
 	 */
 	if (packet_ctx->iv_included) return 0;
 
-	CHECK_FREESPACE(outlen, 4 + AKA_SIM_IV_SIZE);	/* AT_IV + Length + Reserved(2) + IV */
+	FR_DBUFF_CHECK_REMAINING_RETURN(&work_dbuff, 4 + AKA_SIM_IV_SIZE);	/* AT_IV + Length + Reserved(2) + IV */
```

Comment:
This is now redundant because we do length checks everywhere else.

Reply:
Good point. I will get rid of it.

Reply:
Redundant check removed, and the change pushed.